### DISCUSSION TITLE
11ty.js: Render JavaScript in Markdown?

### DISCUSSION THREAD

**Discussion by lkhrs** [11ty/eleventy#2957] [Q&A]
Hi,

I'm working on a project where I'm using 11ty.js as the template engine for Markdown files ([relevant doc section](https://www.11ty.dev/docs/languages/javascript/#markdown-and-javascript)). I'd like to render a value from a global data file inside a Markdown file, like so:

```markdown
[This is a link](${data.metadata.url})
```

Is there a way to render template literals or use 11ty.js shortcodes in Markdown files? I'm aware of `markdown-it-container`, but that's block-level.

**Comment by dnywh** [comment]
You’re close. Just a slight syntax change in your Markdown file:

```md
[This is a link]({{ metadata.url }})
```

This assumes:
- Your data file is called _metadata.json_ or _metadata.js_ and lives inside of `src/_data`
  - This data file is formatted [correctly](https://www.11ty.dev/docs/data-global/)
- You meant to write that you’re using Eleventy to generate your site (which [defaults](https://www.11ty.dev/docs/config/#default-template-engine-for-markdown-files) to `liquid` as the default template engine for Markdown files)

You can use shortcodes in your Markdown files [out-of-the-box](https://www.11ty.dev/docs/shortcodes/).

**Reply by lkhrs** [reply]
Thanks for the response. I know you can do this with Liquid and other supported templating languages, however I'm trying to do this while using JavaScript templates. I have 11ty.js set as my Markdown template engine.
